### PR TITLE
SONAR-8335 fix changing log level through WS

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java [hunk 1]
```diff
@@ -93,53 +92,61 @@ public LoggerContextListener enableJulChangePropagation(LoggerContext loggerCont
...
       }
+    }
+    return newLevel;
+  }
 
-      public Builder setProcessId(ProcessId processId) {
-        this.processId = processId;
-        return this;
-      }
+  private void applyHardcodedLevel(LoggerContext rootContext, String loggerName, Level newLevel) {
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java]: MAJOR Make "applyHardcodedLevel" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/c0b5c1735c7cf1f297ef353eca5735817ccdec2a/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)
1. MAJOR [LogbackHelper.java#L189](https://github.com/SonarSource/sonarqube/blob/c0b5c1735c7cf1f297ef353eca5735817ccdec2a/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L189): Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)
1. MAJOR [LogbackHelper.java#L246](https://github.com/SonarSource/sonarqube/blob/c0b5c1735c7cf1f297ef353eca5735817ccdec2a/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L246): Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)
1. MAJOR [LogbackHelper.java#L273](https://github.com/SonarSource/sonarqube/blob/c0b5c1735c7cf1f297ef353eca5735817ccdec2a/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L273): NullPointerException might be thrown as 'rollingPolicy' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/e0fef1f85f819103dee00f3a3df1e611195120b2/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)
1. MAJOR [LogbackHelper.java#L189](https://github.com/SonarSource/sonarqube/blob/e0fef1f85f819103dee00f3a3df1e611195120b2/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L189): Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)
1. MAJOR [LogbackHelper.java#L246](https://github.com/SonarSource/sonarqube/blob/e0fef1f85f819103dee00f3a3df1e611195120b2/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L246): Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)
1. MAJOR [LogbackHelper.java#L273](https://github.com/SonarSource/sonarqube/blob/e0fef1f85f819103dee00f3a3df1e611195120b2/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L273): NullPointerException might be thrown as 'rollingPolicy' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

File: server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java [hunk 2]
```diff
@@ -268,57 +243,13 @@ private static Level getPropertyValueAsLevel(Props props, String propertyKey) {
 
   private static boolean isAllowed(Level level) {
     for (Level allowedRootLogLevel : ALLOWED_ROOT_LOG_LEVELS) {
-      if (level == allowedRootLogLevel) {
+      if (level.equals(allowedRootLogLevel)) {
```
Comment (SonarTech) [inline server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/35c5761c5f84af5d212440fcb1f5915f6a7acd96/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)
1. MAJOR [LogbackHelper.java#L189](https://github.com/SonarSource/sonarqube/blob/35c5761c5f84af5d212440fcb1f5915f6a7acd96/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L189): Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)
1. MAJOR [LogbackHelper.java#L273](https://github.com/SonarSource/sonarqube/blob/35c5761c5f84af5d212440fcb1f5915f6a7acd96/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L273): NullPointerException might be thrown as 'rollingPolicy' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/35c5761c5f84af5d212440fcb1f5915f6a7acd96/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)
1. MAJOR [LogbackHelper.java#L189](https://github.com/SonarSource/sonarqube/blob/35c5761c5f84af5d212440fcb1f5915f6a7acd96/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L189): Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)
1. MAJOR [LogbackHelper.java#L273](https://github.com/SonarSource/sonarqube/blob/35c5761c5f84af5d212440fcb1f5915f6a7acd96/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L273): NullPointerException might be thrown as 'rollingPolicy' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/375e775d9ad99cf238c5053cf8e8aa8040970efb/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)
1. MAJOR [LogbackHelper.java#L189](https://github.com/SonarSource/sonarqube/blob/375e775d9ad99cf238c5053cf8e8aa8040970efb/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L189): Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)
1. MAJOR [LogbackHelper.java#L273](https://github.com/SonarSource/sonarqube/blob/375e775d9ad99cf238c5053cf8e8aa8040970efb/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L273): NullPointerException might be thrown as 'rollingPolicy' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

File: server/sonar-process/src/main/java/org/sonar/process/logging/LogDomain.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.process.logging;
+
+public enum LogDomain {
+  SQL("sql"),
+  ES_CLIENT("es"),
```
Comment (simonbrandhof) [inline server/sonar-process/src/main/java/org/sonar/process/logging/LogDomain.java]: Minor - constant `ES` sounds enough to me. No need for _CLIENT suffix.

File: server/sonar-process/src/main/java/org/sonar/process/logging/LogLevelConfig.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+  Map<String, List<String>> getConfiguredByProperties() {
+    return configuredByProperties;
+  }
+
+  Map<String, Level> getConfiguredByHardcodedLevel() {
+    return configuredByHardcodedLevel;
+  }
+
+  public static Builder newBuilder() {
```
Comment (simonbrandhof) [inline server/sonar-process/src/main/java/org/sonar/process/logging/LogLevelConfig.java]: this method is useless as Builder constructor is public (which is good enough IMO as it's not an API)

File: server/sonar-server/src/test/java/org/sonar/ce/log/CeLoggingTest.java [hunk 5]
```diff
@@ -75,4 +75,70 @@ public void clearForTask_removes_task_uuid_from_MDC() {
     assertThat(MDC.get(MDC_CE_TASK_UUID)).isNull();
   }
 
+
+//  @Test
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/ce/log/CeLoggingTest.java]: commented-out code

File: server/sonar-process/src/main/java/org/sonar/process/logging/LogLevelConfig.java (hunk 4 above)
Comment (sns-seb) [inline server/sonar-process/src/main/java/org/sonar/process/logging/LogLevelConfig.java]: I'll make the constructor private then, I prefer `LogLevelConfig.newBuilder()` over `new LogLevelConfig.Builder()` and it's 1 char shorter :)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/7ac3b5e9f4fccb2474bdfdb3750c94dd7eee063c/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)
1. MAJOR [LogbackHelper.java#L189](https://github.com/SonarSource/sonarqube/blob/7ac3b5e9f4fccb2474bdfdb3750c94dd7eee063c/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L189): Make the "rootLogger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1312)
1. MAJOR [LogbackHelper.java#L273](https://github.com/SonarSource/sonarqube/blob/7ac3b5e9f4fccb2474bdfdb3750c94dd7eee063c/server/sonar-process/src/main/java/org/sonar/process/logging/LogbackHelper.java#L273): NullPointerException might be thrown as 'rollingPolicy' is nullable here [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Pull request merged**